spec/integration/integration_spec.rb

```diff
@@ -901,5 +901,28 @@ class Tag
         end
       end
     end
+
+    describe 'overriden changes_method with non_database_fields' do
```

Comment:
I think I understand better now. It's not just the fields, it's also what has changed and therefore you have the `changes_method`. 

Now that I think of it I believe this suggestion feels very awkward as implemented. I think I would want a way to make those change fields behave the same way as mongoid fields, aka that you get a `field_was` method that returns the old value and a `field_changed?` method. 

Otherwise for your purposes a much simpler implementation would be to override `history_tracker_attributes(action)` (see https://github.com/aq1018/mongoid-history/blob/master/lib/mongoid/history/trackable.rb#L229) and you can add whatever you want into `:original` and `:modified`.